AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+/// Error Handler function typealias
+public typealias DataStoreErrorHandler = (AmplifyError) -> Void
+
+/// Holds a reference to both the local `Model` and the remote one during a conflict
+/// resolution. Implementations of the `DataStoreConflictHandler` use this to decide
+/// what the outcome of a conflict should be.
+public struct DataStoreConclictData {
+    public let local: Model
+    public let remote: Model
```

Comment:
these have to be `MutationSync` for the developer to check if the local/remote was deleted

Reply:
Lets handle this in a subsequent PR

Reply:
i agree, currently i've made changes like
```
public struct DataStoreConclictData {
    public let local: LocalModelData
    public let remote: RemoteModelData
}

public enum LocalModelData {
    case update(Model)
    case delete(Model.Identifier)
}

public enum RemoteModelData {
    case update(Model)
    case delete(Model)
}
```